File: backend/content-service/src/main/java/com/communityplatform/content/service/LikeService.java
package com.communityplatform.content.service;

import com.communityplatform.content.dto.like.LikeResponseDto;

import java.util.List;

/**
 * Service interface for Like operations.
 */
public interface LikeService {

    /**
     * Like a post.
     */
    LikeResponseDto likePost(Long postId, Long userId);

    /**
     * Unlike a post.
     */
    void unlikePost(Long postId, Long userId);

    /**
     * Like a comment.
     */
    LikeResponseDto likeComment(Long commentId, Long userId);

    /**
     * Unlike a comment.
     */
    void unlikeComment(Long commentId, Long userId);

    /**
     * Check if user has liked a post.
     */
    boolean hasLikedPost(Long postId, Long userId);

    /**
     * Check if user has liked a comment.
     */
    boolean hasLikedComment(Long commentId, Long userId);

    /**
     * Get all likes for a post.
     */
    List<LikeResponseDto> getPostLikes(Long postId);

    /**
     * Get all likes for a comment.
     */
    List<LikeResponseDto> getCommentLikes(Long commentId);
}

File: frontend/src/app/core/services/like.service.ts
import { inject, Injectable } from '@angular/core';
import { ApiService } from './api.service';
import { LikeResponse } from '../models/like.model';

@Injectable({ providedIn: 'root' })
export class LikeService {
  private api = inject(ApiService);

  likePost(postId: number) {
    return this.api.post<void>(`/v1/posts/${postId}/like`);
  }

  unlikePost(postId: number) {
    return this.api.delete<void>(`/v1/posts/${postId}/like`);
  }

  hasLikedPost(postId: number) {
    return this.api.get<boolean>(`/v1/posts/${postId}/like/check`);
  }

  getPostLikes(postId: number) {
    return this.api.get<LikeResponse[]>(`/v1/posts/${postId}/likes`);
  }

  likeComment(commentId: number) {
    return this.api.post<void>(`/v1/comments/${commentId}/like`);
  }

  unlikeComment(commentId: number) {
    return this.api.delete<void>(`/v1/comments/${commentId}/like`);
  }

  hasLikedComment(commentId: number) {
    return this.api.get<boolean>(`/v1/comments/${commentId}/like/check`);
  }

  getCommentLikes(commentId: number) {
    return this.api.get<LikeResponse[]>(`/v1/comments/${commentId}/likes`);
  }
}
